dynamic_retriever.py

```diff
@@ -0,0 +1,47 @@
+from langchain.chains.router import MultiRetrievalQAChain
+from langchain.llms import OpenAI
+
+from langchain.embeddings import OpenAIEmbeddings
+from langchain.document_loaders import TextLoader
+from langchain.vectorstores import FAISS
+
+sou_docs = TextLoader('../../state_of_the_union.txt').load_and_split()
+sou_retriever = FAISS.from_documents(sou_docs, OpenAIEmbeddings()).as_retriever()
+
+pg_docs = TextLoader('../../paul_graham_essay.txt').load_and_split()
+pg_retriever = FAISS.from_documents(pg_docs, OpenAIEmbeddings()).as_retriever()
+
+personal_texts = [
+    "I love apple pie",
+    "My favorite color is fuchsia",
+    "My dream is to become a professional dancer",
+    "I broke my arm when I was 12",
+    "My parents are from Peru",
+]
+personal_retriever = FAISS.from_texts(personal_texts, OpenAIEmbeddings()).as_retriever()
+
+retriever_infos = [
+    {
+        "name": "state of the union", 
+        "description": "Good for answering questions about the 2023 State of the Union address", 
+        "retriever": sou_retriever
+    },
+    {
+        "name": "pg essay", 
+        "description": "Good for answering questions about Paul Graham's essay on his career",
+        "retriever": pg_retriever
+    },
+    {
+        "name": "personal", 
+        "description": "Good for answering questions about me", 
+        "retriever": personal_retriever
+    }
+]
+
+chain = MultiRetrievalQAChain.from_retrievers(OpenAI(), retriever_infos, verbose=True)
+
+print(chain.run("What did the president say about the economy?"))
+
+print(chain.run("What is something Paul Graham regrets about his work?"))
+
+print(chain.run("What is my background?"))
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

While the code seems fine, it's important to note that error handling is missing. If any of the `chain.run()` calls fail (for example, due to an issue with the model or the retriever), the entire script will stop execution. It would be better to wrap these calls in a try-except block and handle potential exceptions appropriately.
```python
try:
    print(chain.run("What did the president say about the economy?"))
except Exception as e:
    print(f"Error while processing query: {e}")

try:
    print(chain.run("What is something Paul Graham regrets about his work?"))
except Exception as e:
    print(f"Error while processing query: {e}")

try:
    print(chain.run("What is my background?"))
except Exception as e:
    print(f"Error while processing query: {e}")
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->